include/seqan3/alphabet/quality/composition.hpp

```diff
@@ -0,0 +1,171 @@
+ * alphabet.
+ *
+ * As with all alphabet_composition s you may access the individual alphabet letters in
+ * regular c++ tuple notation, i.e. `std::get<0>(t)` and objects can be brace-initialized
+ * with the individual members.
+ *
+ * ~~~~~~~~~~~~~~~{.cpp}
+ *
+ * quality_composition<dna4, illumina18> l{dna4::A, 7};
+ * std::cout << int(to_integral(l)) << ' '
+ *           << int(to_integral(std::get<0>(l))) << ' '
+ *           << int(to_integral(std::get<1>(l))) << '\n';
+ * // 148 0 7
+ *
+ * std::cout << to_char(l) << ' '
+ *           << to_char(std::get<0>(l)) << ' '
+ *           << to_char(std::get<1>(l)) << '\n';
+ * // A A (
+ *
+ * std::cout << int(to_phred(l)) << ' '
+ * //           << int(to_phred(std::get<0>(l))) << ' ' // dna4 doesn't have a phred
+ *           << int(to_phred(std::get<1>(l))) << '\n';
+ * // 7 7
+ *
+ * // modify via structured bindings and references:
+ * auto & [ seq_l, qual_l ] = l;
+ * seq_l = dna4::G;
+ * std::cout << to_char(l) << '\n';
+ * // G
+ *
+ * ~~~~~~~~~~~~~~~
+ *
+ * This alphabet_composition itself fulfills both seqan3::alphabet_concept and seqan3::quality_concept .
+ */
+
+template <typename sequence_alphabet_t, typename quality_alphabet_t>
+      requires alphabet_concept<sequence_alphabet_t> &&
+               quality_concept<quality_alphabet_t>
+struct quality_composition :
+    public alphabet_composition<quality_composition<sequence_alphabet_t, quality_alphabet_t>, sequence_alphabet_t, quality_alphabet_t>
+{
+    //!\brief First template parameter as member type.
+    using sequence_alphabet_type = sequence_alphabet_t;
+    //!\brief Second template parameter as member type.
+    using quality_alphabet_type = quality_alphabet_t;
+
+    //!\brief Equals the char_type of sequence_alphabet_type.
+    using char_type = underlying_char_t<sequence_alphabet_type>;
+    //!\brief Equals the phred_type of the quality_alphabet_type.
+    using phred_type = underlying_phred_t<quality_alphabet_type>;
+
+    /*!\name Write functions
+     * \{
+     */
+    //!\brief Directly assign the sequence letter.
+    constexpr quality_composition & operator=(sequence_alphabet_type const l)
+    {
+        std::get<0>(*this) = l;
+        return *this;
+    }
+
+    //!\brief Directly assign the quality letter.
+    constexpr quality_composition & operator=(quality_alphabet_type const l)
+    {
+        std::get<1>(*this) = l;
+        return *this;
+    }
+
+    //!\brief Assign from a character. This modifies the internal sequence letter.
+    constexpr quality_composition & from_char(char_type const c)
+    {
+        seqan3::from_char(std::get<0>(*this), c);
+        return *this;
+    }
+
+    //!\brief Assign from a phred value. This modifies the internal quality letter.
+    constexpr quality_composition & from_phred(phred_type const c)
+    {
+        seqan3::from_phred(std::get<1>(*this), c);
+        return *this;
+    }
```

Comment:
the assignment ops: yes, I have added it now.
the from_*: no, these do bounds-checking in debug mode.